ui/server/src/main/scala/pl/touk/nussknacker/ui/process/deployment/DeploymentService.scala

```diff
@@ -61,34 +61,30 @@ class DeploymentService(processRepository: FetchingProcessRepository[Future],
                     performDeploy: (ProcessingType, ProcessVersion, DeploymentData, GraphProcess, Option[String]) => Future[_])
                    (implicit user: LoggedUser): Future[ProcessActionEntityData] = {
     for {
-      processingType <- processRepository.fetchProcessingType(processId)
-      maybeLatestVersion <- processRepository.fetchLatestProcessVersion[DisplayableProcess](processId)
-      latestVersion <- processDataExistOrFail(maybeLatestVersion, processId.value.toString)
-      result <- deployAndSaveProcess(processingType, latestVersion, savepointPath, comment, performDeploy)
+      maybeProcess <- processRepository.fetchLatestProcessDetailsForProcessId[CanonicalProcess](processId)
```

Comment:
`fetchLatestProcessDetailsForProcessId` is more expensive, we really need it here?